core/test/variable_test.cpp

```diff
@@ -8,6 +8,7 @@
 #include "scipp/core/dimensions.h"
 #include "scipp/core/except.h"
 #include "scipp/core/variable.h"
+#include "scipp/core/variable.tcc"
```

Comment:
I would avoid having the `tcc` included in the main test file. After all we want to ensure that the built-in code for `Variable` compiles and runs *without* having that file included.

Move the instantiation tests to a different source file?